File: backend/src/service/bayesian_service.py
from src.model.bayesian import BayesianModel

class BayesianService:
    def __init__(self, model=None):
        self.model = model or BayesianModel()

    def get_network(self):
        """Fetch Bayesian Network data."""
        nodes = self.model.fetch_nodes()
        edges = self.model.fetch_edges()
        return {"nodes": nodes, "edges": edges}

File: front/src/api/services/bayesianService.ts
import apiClient from "../apiClient";

export const getBayesianNetworkStatus = async () => {
    try {
        const response = await apiClient.get("/bayesian/");
        return response.data;
    } catch (error) {
        console.error("Failed to fetch Bayesian network status:", error);
        throw error;
    }
};
